src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -144,41 +144,39 @@ private void OnSymbolEnd(SymbolAnalysisContext context)
                     return;
                 }
 
-                var message = GetMessageForUnusedParameterDiagnostic(
-                    parameter.Name,
+                var rule = GetDescriptorForUnusedParameterDiagnostic(
                     hasReference,
                     isPublicApiParameter: parameter.ContainingSymbol.HasPublicResultantVisibility(),
                     isLocalFunctionParameter: parameter.ContainingSymbol.IsLocalFunction());
 
-                var diagnostic = DiagnosticHelper.CreateWithMessage(s_unusedParameterRule, location,
-                    option.Notification.Severity, additionalLocations: null, properties: null, message);
-                reportDiagnostic(diagnostic);
+                var diagnostic = DiagnosticHelper.Create(rule, location,
```

Comment:
It would be much cleaner to rename this method to `TryCreate` and have it return a `Diagnostic?`. Also you can easily avoid the null check at each callsite by having a helper `TryCreateAndReport(..., reportDiagnostic)`.

Reply:
@mavasani My intent was to avoid too many changes to analyzers.